libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java

```diff
@@ -287,6 +286,58 @@ public SyncState syncUp(SyncUpTarget target, SyncOptions options, String soupNam
     	return sync;
     }
 
+    /**
+     * Deletes local copies of records that have been deleted on the server.
+     *
+     * @param syncId Sync ID.
+     * @throws JSONException
+     */
+    public void syncRemoteDeletes(long syncId) throws JSONException {
+        if (runningSyncIds.contains(syncId)) {
+            throw new SmartSyncException("Cannot run syncRemoteDeletes:" + syncId + ": still running");
+        }
+        final SyncState sync = SyncState.byId(smartStore, syncId);
+        if (sync == null) {
+            throw new SmartSyncException("Cannot run syncRemoteDeletes:" + syncId + ": no sync found");
+        }
+        if (sync.getType() != SyncState.Type.syncDown) {
+            throw new SmartSyncException("Cannot run syncRemoteDeletes:" + syncId + ": wrong type:" + sync.getType());
+        }
+        final String soupName = sync.getSoupName();
+        final String idFieldName = sync.getTarget().getIdFieldName();
+
+        /*
+         * Fetches list of IDs present in local soup that have not been modified locally.
+         */
+        final Set<String> localIds = new HashSet<String>();
+        QuerySpec querySpec = QuerySpec.buildAllQuerySpec(soupName, idFieldName, QuerySpec.Order.ascending, 10);
+        int count = smartStore.countQuery(querySpec);
+        querySpec = QuerySpec.buildSmartQuerySpec("SELECT {" + soupName + ":" + idFieldName + "} FROM {" + soupName + "} WHERE {" + soupName + ":" + LOCAL + "}='false'", count);
+        final JSONArray localIdArray = smartStore.query(querySpec, 0);
+        if (localIdArray != null && localIdArray.length() > 0) {
+            for (int i = 0; i < localIdArray.length(); i++) {
+                final JSONArray idJson = localIdArray.optJSONArray(i);
+                if (idJson != null) {
+                    localIds.add(idJson.optString(0));
+                }
+            }
+        }
+
+        /*
+         * Fetches list of IDs still present on the server from the list of local IDs
+         * and removes the list of IDs that are still present on the server.
+         */
+        final Set<String> remoteIds = ((SyncDownTarget) sync.getTarget()).getListOfRemoteIds(this, localIds);
```

Comment:
At this point, we know it has to be a sync down operation, so we can safely assume so here to get remote IDs.